### PR TITLE
Add IRegionOrigin and AffixedRibbon fields to OT/Misc tab

### LINKED ISSUE #417: Add IRegionOrigin fields and Affixed Ribbon to OT/Misc tab
## Summary

Two field groups are missing from the **OT/Misc tab** of the Pokémon editor.

### Country / Sub-Region / Console Region (`IRegionOrigin`, Gen 6–7 only)
- Three dropdowns representing the native 3DS geographic origin of the Pokémon: `Country`, `Region` (sub-region), and `ConsoleRegion`.
- Applies to: PK6 (X/Y, ORAS), PK7 (SM, USUM). Not present in Gen 8+.
- **Note:** This is distinct from the five `IGeoTrack` visited-country slots, which are already implemented. `IRegionOrigin` is the Pokémon's *native* 3DS region, fixed at time of capture.
- PKHeX reference: `CB_Country`, `CB_SubRegion`, `CB_3DSReg` in `PKMEditor.cs`.
- Country/region data: `Util.GetCountryRegionList("countries", ...)` and `Util.GetCountryRegionList("sr_" + country, ...)`.

### Affixed Ribbon / Mark (`PKM.AffixedRibbon`, Gen 8+)
- A selector showing which ribbon or mark is currently displayed on the Pokémon's summary screen in-game.
- Stored as a `RibbonIndex` enum value (`-1` = none).
- Applies to: PK8, PB8, PA8, PK9, PA9.
- PKHeX reference: `PB_Affixed` (clickable picture box that opens a ribbon picker) and `AffixedTip` tooltip in `PKMEditor.cs`.

## Implementation Notes
- `IRegionOrigin` fields should only render when `pk is IRegionOrigin`.
- `AffixedRibbon` field should only render when `pk is IRibbonIndex` (or when `PKM.AffixedRibbon` is accessible, Gen 8+).
- For the affixed ribbon picker, reuse the existing ribbon list from the Ribbons tab where possible.

## Roadmap
See `PKHEX_PARITY_ROADMAP.md` §1.1 — OT/Misc Tab — Missing Fields.

### PR DESCRIPTION
**Pull request by codemonkey85**
Closes #417

## Summary

- **Country / Sub-Region / Console Region** (`IRegionOrigin`, Gen 6–7): Three new dropdowns added to the OT/Misc tab below the Geo Locations section. Country and Sub-Region reuse the existing `GetGeoCountryComboItems` / `GetGeoRegionComboItems` helpers (same data source as Geo Locations). Sub-Region resets to 0 when Country changes. Console Region uses a new `GetConsoleRegionComboItems()` backed by `GameInfo.FilteredSources.ConsoleRegions` (the `Region3DSIndex` enum values — Japan, North America, Europe, China, Korea, Taiwan).
- **Affixed Ribbon / Mark** (`IRibbonSetAffixed`, Gen 8+): A new dropdown at the end of the OT/Misc tab. Populated with "None" (value -1) plus all ribbons and marks the Pokémon currently holds, using `RibbonHelper.GetAllRibbonInfo`. If the currently-affixed value is absent from the ribbon list (edge case), it is still shown with a `(!)` marker. The section only renders when the Pokémon implements `IRibbonSetAffixed` (PK8, PB8, PA8, PK9, PA9).

## Test plan

- [ ] Open a Gen 6 or Gen 7 Pokémon (PK6 / PK7) → OT/Misc tab shows "Native 3DS Region" section with Country, Sub-Region, and Console Region dropdowns
- [ ] Changing Country resets Sub-Region to "—"
- [ ] Sub-Region dropdown is hidden when Country is "—" (0)
- [ ] Open a Gen 8+ Pokémon (PK8/PB8/PA8/PK9/PA9) → OT/Misc tab shows "Affixed Ribbon / Mark" dropdown
- [ ] Dropdown lists only ribbons/marks the Pokémon has plus "None"; selecting one updates the value; selecting "None" sets it to −1
- [ ] Gen 5 or earlier Pokémon: neither section is visible
- [ ] Build passes with zero warnings: `dotnet build Pkmds.Web/Pkmds.Web.csproj -c Debug`
- [ ] Tests pass: `dotnet test`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Pkmds.Rcl/Components/EditForms/Tabs/OtMiscTab.razor.cs [hunk 1]
```diff
@@ -381,4 +381,59 @@ private void SetPokemonHomeTracker(string newTrackerHex)
...
+        {
+            var has = info.Type == RibbonValueType.Boolean ? info.HasRibbon : info.RibbonCount > 0;
+            if (!has)
+            {
+                continue;
+            }
+
+            var shortName = info.Name.StartsWith("Ribbon", StringComparison.Ordinal)
+                ? info.Name["Ribbon".Length..]
+                : info.Name;
```
[the remark this answers is not in the document]
Comment (codemonkey85) [reply Pkmds.Rcl/Components/EditForms/Tabs/OtMiscTab.razor.cs]: Fixed — the  prefix is now stripped before falling back to , so count-based ribbon properties (e.g. `RibbonCountMemoryContest` → `MemoryContest`) correctly resolve to a `RibbonIndex` entry. The Boolean-only type filter was removed; `Enum.TryParse` acts as the natural gate.

File: Pkmds.Rcl/Components/EditForms/Tabs/OtMiscTab.razor.cs [hunk 2]
```diff
@@ -381,4 +381,59 @@ private void SetPokemonHomeTracker(string newTrackerHex)
...
+            var displayName = RibbonHelper.GetRibbonDisplayName(info.Name);
+            items.Add(new ComboItem(displayName, (int)idx));
+            added.Add((int)idx);
+        }
+
+        // Ensure the currently affixed ribbon is in the list even if the Pokémon doesn't have it
+        if (currentAffixed != PKHeX.Core.AffixedRibbon.None && !added.Contains((int)currentAffixed))
+        {
+            var idx = (RibbonIndex)currentAffixed;
+            var propName = $"Ribbon{idx}";
```
[the remark this answers is not in the document]
Comment (codemonkey85) [reply Pkmds.Rcl/Components/EditForms/Tabs/OtMiscTab.razor.cs]: No longer applicable — the fallback block (for a currently-affixed ribbon not in the dropdown list) was removed in a prior commit. `BuildAffixedRibbonItems` now includes all ribbon/mark slots supported by the Pokémon's format unconditionally, so the fallback is never needed.

File: Pkmds.Rcl/Components/EditForms/Tabs/OtMiscTab.razor [hunk 3]
```diff
@@ -650,5 +650,108 @@
             </MudItem>
         }
 
+        @* Native 3DS Region (Gen 6–7, IRegionOrigin) *@
+        @if (Pokemon is IRegionOrigin regionOrigin)
+        {
+            var countries = AppService.GetGeoCountryComboItems().ToList();
```
[the remark this answers is not in the document]
Comment (codemonkey85) [reply Pkmds.Rcl/Components/EditForms/Tabs/OtMiscTab.razor]: Fixed — removed the redundant `.ToList()` so the `IRegionOrigin` block enumerates `GetGeoCountryComboItems()` directly, avoiding a second materialisation when the Pokémon also implements `IGeoTrack`.

File: Pkmds.Rcl/Services/AppService.cs [hunk 4]
```diff
@@ -844,6 +844,9 @@ public IEnumerable<ComboItem> GetGeoRegionComboItems(byte country)
         return items;
     }
 
+    public IReadOnlyList<ComboItem> GetConsoleRegionComboItems() =>
+        GameInfo.FilteredSources.ConsoleRegions;
```
[the remark this answers is not in the document]
Comment (codemonkey85) [reply Pkmds.Rcl/Services/AppService.cs:848]: Added `GetConsoleRegionComboItems_ReturnsExpectedRegions` to `AppServiceTests` — asserts the result is non-empty, all items have non-empty text, and the three core `Region3DSIndex` values (Japan, NorthAmerica, Europe) are present.
